Sources/AblyChat/ChatAPI.swift

```diff
@@ -0,0 +1,139 @@
+import Ably
+
+public final class ChatAPI: Sendable {
+    private let realtime: RealtimeClient
+    private let apiVersion = "/chat/v1"
+
+    public init(realtime: RealtimeClient) {
+        self.realtime = realtime
+    }
+
+    internal func getChannel(_ messagesChannelName: String) -> any RealtimeChannelProtocol {
+        realtime.getChannel(messagesChannelName)
+    }
+
+    // (CHA-M6) Messages should be queryable from a paginated REST API.
+    public func getMessages(roomId: String, params: QueryOptions) async throws -> any PaginatedResult<Message> {
+        let endpoint = "\(apiVersion)/rooms/\(roomId)/messages"
+        return try await makeAuthorizedPaginatedRequest(endpoint, params: params.toDictionary())
+    }
+
+    internal struct SendMessageResponse: Codable {
+        internal let timeserial: String
+        internal let createdAt: Int64
+    }
+
+    // (CHA-M3) Messages are sent to Ably via the Chat REST API, using the send method.
+    // (CHA-M3a) When a message is sent successfully, the caller shall receive a struct representing the Message in response (as if it were received via Realtime event).
+    public func sendMessage(roomId: String, params: SendMessageParams) async throws -> Message {
+        let endpoint = "\(apiVersion)/rooms/\(roomId)/messages"
+        var body: [String: Any] = ["text": params.text]
+
+        // (CHA-M3b) A message may be sent without metadata or headers. When these are not specified by the user, they must be omitted from the REST payload.
+        if let metadata = params.metadata {
+            body["metadata"] = metadata
+
+            // (CHA-M3c) metadata must not contain the key ably-chat. This is reserved for future internal use. If this key is present, the send call shall terminate by throwing an ErrorInfo with code 40001.
+            if metadata.contains(where: { $0.key == "ably-chat" }) {
+                throw ARTErrorInfo.create(withCode: 40001, message: "metadata must not contain the key `ably-chat`")
+            }
+        }
+
+        if let headers = params.headers {
+            body["headers"] = headers
+
+            // (CHA-M3d) headers must not contain a key prefixed with ably-chat. This is reserved for future internal use. If this key is present, the send call shall terminate by throwing an ErrorInfo with code 40001.
+            if headers.keys.contains(where: { keyString in
+                keyString.hasPrefix("ably-chat")
+            }) {
+                throw ARTErrorInfo.create(withCode: 40001, message: "headers must not contain any key with a prefix of `ably-chat`")
+            }
+        }
+
+        let response: SendMessageResponse = try await makeAuthorizedRequest(endpoint, method: "POST", body: body)
+
+        // response.createdAt is in milliseconds, convert it to seconds
+        let createdAtInSeconds = TimeInterval(integerLiteral: response.createdAt / 1000)
+
+        let message = Message(
```

Comment:
It's from the spec (also commented directly above the function)

>     // (CHA-M3a) When a message is sent successfully, the caller shall receive a struct representing the Message in response (as if it were received via Realtime event).

The API response here only returns the `timeserial` and `createdAt`, so we have to construct the rest of the `Message` to fulfil this requirement. Chat-JS achieves the same result but passes in ClientId to `DefaultMessages` and then uses that instead of directly using the clientId from Realtime.

Reply:
hmm i'm not convinced given we have that requirement to init `Realtime` with a `clientId`. Makes sense to use the `clientId` on the `Realtime` instance whenever we need a `clientId`?

Reply:
> Makes sense to use the clientId on the Realtime instance whenever we need a clientId?

Sure, but IMO "makes sense" isn't the same as "is obvious and doesn't need specifying" here

Reply:
this seems more like a spec issue, let's raise it there? I think it's clear from looking at the code where clientId could/should/does come at the moment so it shouldn't hold this PR up

Reply:
I’ll leave it to you to raise it